File: scripts/c/factorial.c
/*To find the Factorial of a Number*/
#include<stdio.h>
int factorial(int);

int main()
{
 int n,fact;
 printf("Enter a number: \n");
 scanf("%d",&n);
 fact=factorial(n);
 printf("Factorial of the number is %d \n",fact);
}

int factorial(int x)
{
 int f;
 if(x==0)
 f=1;
 else
 f=x*factorial(x-1);
 return(f);
}

File: scripts/java/factorial.java
/*factorial of a number */
import java.io.*;

class factorial
{
 public static void main(String args[]) throws IOException
 {
  int n,i,fact=1;
  System.out.println(" Enter the n value ");
  DataInputStream dis=new DataInputStream(System.in);
  n=Integer.parseInt(dis.readLine());
  if(n==0)
  {
   System.out.println("The Factorial of "+n+" is 1");
  }
  else
  {
    for(i=1;i<=n;i++)
	  { 
	   fact=fact*i;
	  }
    System.out.println("The Factorial of "+n+" is "+fact);	
  }  
 }
}
